server/core/src/actors/internal.rs

```diff
@@ -100,22 +100,55 @@ impl QueryServerWriteV1 {
         }
     }
 
-    pub(crate) async fn handle_delayedaction(&self, da: DelayedAction) {
+    pub(crate) async fn handle_delayedaction(&self, da_batch: &mut Vec<DelayedAction>) {
```

Comment:
Since we don't use the batch anywhere else we don't really need the mut and can take ownership of it, by the looks of it?

Reply:
We re-use the vec as a buffer here.